addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1057,6 +1057,9 @@ export class PosStore extends Reactive {
                 // a higher id than the original one since it's the last one created.
                 const session = this.models["pos.session"].sort((a, b) => a.id - b.id)[0];
                 session.delete();
+                this.models["pos.order"].getAll()[0].session_id = this.models["pos.session"].sort(
+                    (a, b) => a.id - b.id
+                )[0];
```

Comment:
Almost there :smiley:  This is what I used for a small test where I knew I would only get 1 order.  It needed to be adapted and cleaned up a bit (cause I really only wrote that to test my idea). 

Here you only assign the first order, we need to do that for all instead because you might have more than 1 order. To clean up I think you can just use `this.models["pos.session"].getFirst()` for example

Reply:
Hey @sbel-odoo 

I think this is more clean 😄 
What do you think?